dev_guide/builds.adoc

```diff
@@ -88,8 +99,9 @@ image tag or the source code changes:
 *ruby-sample-build*.
 <2> You can specify a list of link:#build-triggers[triggers], which cause a new
 build to be created.
-<3> The `*source*` section defines the source code repository location. You can
-provide additional options, such as `*sourceSecret*` or `*contextDir*` here.
+<3> The `*source*` section defines the source of the build. The source type can
+be either `*Git*`, to point to a code repository location, or `*Dockerfile*`, to
+build from an inline Dockerfile.
```

Comment:
which points to an issue with the current BuildSource api definition because it demands a BuildSourceType value to be "Git", "Dockerfile", or "Binary" but in reality if you specific "Git" you can *also* provide a Dockerfile (which will replace the Dockerfile in the Git repository, if there is one).

Really we should get rid of the BuildSourceType and update our code to reflect the fact that you can have multiple simultaneous Source inputs.  I think @smarterclayton has pushed for this before and i'm coming around to it.